enterprise/coderd/workspaceproxy.go

```diff
@@ -208,3 +211,96 @@ func (api *API) workspaceProxyIssueSignedAppToken(rw http.ResponseWriter, r *htt
 		SignedTokenStr: tokenStr,
 	})
 }
+
+// reconnectingPTYSignedToken issues a signed app token for use when connecting
+// to the reconnecting PTY websocket on an external workspace proxy. This is set
+// by the client as a query parameter when connecting.
+//
+// @Summary Issue signed app token for reconnecting PTY
+// @ID issue-signed-app-token-for-reconnecting-pty
+// @Security CoderSessionToken
+// @Tags Applications Enterprise
+// @Accept json
+// @Produce json
+// @Param request body codersdk.IssueReconnectingPTYSignedTokenRequest true "Issue reconnecting PTY signed token request"
+// @Success 200 {object} codersdk.IssueReconnectingPTYSignedTokenResponse
+// @Router /applications/reconnecting-pty-signed-token [post]
+// @x-apidocgen {"skip": true}
+func (api *API) reconnectingPTYSignedToken(rw http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	apiKey := httpmw.APIKey(r)
+	if !api.Authorize(r, rbac.ActionCreate, apiKey) {
+		httpapi.ResourceNotFound(rw)
+		return
+	}
+
+	var req codersdk.IssueReconnectingPTYSignedTokenRequest
+	if !httpapi.Read(ctx, rw, r, &req) {
+		return
+	}
+
+	u, err := url.Parse(req.URL)
+	if err != nil {
+		httpapi.Write(ctx, rw, http.StatusBadRequest, codersdk.Response{
+			Message: "Invalid URL.",
+			Detail:  err.Error(),
+		})
+		return
+	}
```

Comment:
Might want to check `Scheme` is `http` or `https`. Otherwise if you use ports, it messes up the host as the scheme:

https://go.dev/play/p/Tsa8q9jGA4l